guides/common/modules/proc_installing-puppet-agent-during-host-provisioning.adoc

```diff
@@ -1,30 +1,55 @@
 [id="Installing_Puppet_Agent_during_Host_Provisioning_{context}"]
 = Installing Puppet Agent during Host Provisioning
 
-Install the Puppet agent during the host provisioning process.
+Use this procedure to install the Puppet agent during the host provisioning process.
+
+ifdef::satellite[]
+.Prerequisites
+* You enabled and synchronized the *{project-client-name}* repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the *{project-client-name}* repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
+ifdef::katello,orcharhino[]
+.Prerequisites
+* You created a Product and repository for the upstream Puppet agent, such as `\https://yum.puppet.com` or `\https://apt.puppet.com`, and synchronized the repository to {Project}.
+For more information, see {ContentManagementDocURL}Importing_Content_content-management[Importing Content] in _{ContentManagementDocTitle}_.
+* You created an activation key that enables the Puppet agent repository for hosts.
+For more information, see {ContentManagementDocURL}Managing_Activation_Keys_content-management[Managing Activation Keys] in _{ContentManagementDocTitle}_.
+endif::[]
 
 .Procedure
-. Navigate to *Hosts > Provisioning Templates*.
+. Navigate to *Hosts* > *Provisioning Templates*.
 . Select a provisioning template depending on your host provisioning method.
 For more information, see {ProvisioningDocURL}types-of-provisioning-templates_provisioning[Types of Provisioning Templates] in _{ProvisioningDocTitle}_.
+ifndef::katello,orcharhino,satellite[]
+. Ensure the `puppetlabs_repo` snippet is included as follows:
```

Comment:
Open question: I see it's copying what was there before with `puppet_setup`, but I wonder if it's clear to users that the default provisioning templates we ship (kickstart, preseed, etc) already do this. That is actually the reason we tell users to set the host parameter `enable-puppetlabs-repo`: it ensures the snippet gets loaded. So it feels like a redundant instruction.

Reply:
`enable-puppetlabs-repo` ensures both the snippet gets loaded in the default templates AND is also used inside the snippet.
But if we assume that the user uses the default provisioning templates, then both snippet instructions are redundant. Perhaps we should remove the whole part about editing a template?

Reply:
Do we want to remove the section or should we merge this and continue if needed in a follow-up? @Lennonka @ekohl

Reply:
I was also thinking about adding a note to each snippet step that it's already included in the shipped templates. I'll do that instead removing the steps.